### ISSUE TITLE
Provide a link from gherkin feature file to the bpmn file so that on click it opens the diagram

### ISSUE THREAD

**Issue by yurecz** [bpmn-io/vs-code-bpmn-io#102]
__Is your feature request related to a problem? Please describe__

Is it possible to open from the vs-code text file (in my case Gherkin feature file) the BPMN diagram (modeler) also in the vs-code via some kind of link?
The BPMN diagram plays the role of clarification what is happening inside the Gherkin feature file from a process perspective.

I am doing the integration/e2e process tests (not UI tests) using the tool called cucumber: https://cucumber.io/.
It accepts as the input the flat file in the Gherkin format. The flat file can handle the text description of the business process under test but cannot be used to visualize the process flow for better understanding. The process flows are visualized using the bpmn-io plugin for the vs-code editor from this repository.

So in short there are two files: one gherkin feature file and one BPMN XML file. Both are maintained using vs-code. Is any way to add to the flat feature file the link so that the vs-code editor navigate from the feature file to the BPMN model. It should simply open it in the new tab.

__Describe the solution you'd like__

When opening a Gherkin feature file, make it possible to directly open the referenced BPMN file in another tab.

__Describe alternatives you've considered__

My current workaround is that I just always create the diagram file with the same name as the feature file and manually open the BPMN Modeler from the vs-code menu.

__Additional context__

/

**Comment by pinussilvestrus** [comment]
Hi, thanks for your feature request!

To understand your use case a bit better, can you please give some more details about it? Especially by using our [issue template](https://github.com/bpmn-io/vs-code-bpmn-io/issues/new?assignees=&labels=enhancement&template=FEATURE_REQUEST.md).

What is a "Gherkin feature file"? How does your environment look like? Where do you want to open the BPMN file?

**Comment by yurecz** [comment]
Hello,

I am doing the integration/e2e process tests **(not UI tests)** using the tool called cucumber: https://cucumber.io/.
It accepts as the input the flat file in the Gherkin format. The flat file can handle the text description of the business process under test but cannot be used to visualize the process flow for better understanding. The process flows are visualized using the bpmn-io plugin for the vs-code editor from this repository.

So in short there are two files: one gherkin feature file and one BPMN XML file. Both are maintained using vs-code. Is any way to add to the flat feature file the link so that the vs-code editor navigate from the feature file to the BPMN model. It should simply open it in the new tab?

My current workaround is that I just always create the diagram file with the same name as the feature file and manually opens the BPMNM documentation from the vs-code menu.

Best regards,
Yuri

**Comment by pinussilvestrus** [comment]
Thanks for your explanations! I will adjust the issue description accordingly. Can you maybe add an example file for these Gherkin formats?

 I think to achieve what you describe it would need an [additional extension point](https://github.com/bpmn-io/vs-code-bpmn-io/blob/master/package.json#L23) for the Gherkin format to create such links to the referenced BPMN file. That's not trivial I guess but possible in theory.

I will move this to the backlog since I don't expect that we will work on it soon.

**Comment by nikku** [comment]
Closing this issue as it is missing clear specification. With #67 you will be able to open BPMN files out of the box, from within VSCode.
